### DISCUSSION TITLE
DietPi-Software | n8n - Workflow Automation Tool

### DISCUSSION THREAD

**Discussion by berndverhofstadt** [MichaIng/DietPi#5261] [Software Request]
#### Formal software information
- Software title | n8n
- Short description | Workflow Automation Tool
- Official URL (if available) | https://n8n.io/
- GitHub URL (if available) | https://github.com/n8n-io/n8n
- (Official) install documentation | https://docs.n8n.io/getting-started/installation/#self-hosting-n8n

#### Are there similar/alternative software titles available with DietPi-Software?
No.
- ...

#### What makes your requested software better than the above solutions, if available?
It is fully open source, without any (feature) limitation to free users. It has many out of the box connectors which can be used to automate tasks between (cloud) solutions

#### How can DietPi make the installation easier or compatible, than following the install instructions or do APT installation, if available?
It seems to be very simple to install, however I believe that is an added value in the software portfolio of DietPi. The software is very broad and can interact with different solutions.
- ...

#### Can you provide the installation steps that you would suggest DietPi-Software to do?
1. Make sure that npm is available
2. npm install n8n -g

**Comment by MichaIng** [comment]
Many thanks for your suggestion.

Lol, the first question I had in mind answered here: https://github.com/n8n-io/n8n#what-does-n8n-mean-and-how-do-you-pronounce-it
I first thought "nAutomation" like for node-automation, but that would have been 9 characters 😄.

Second question I had in mind is discussed here: https://discourse.nodered.org/t/how-does-node-red-compare-to-n8n-io/37297
But of course, since it's the Node-RED forum, it's not objective/biased. On GitHub, n8n has notably more stars than Node-RED and shows more activity as well. But like with the smart home/automation software options, while they do overlap, they have a different focus, hence are not redundant, I'd say.

Might be interesting for @StephanStS to have a look at, who works a lot with Node-RED.

**Reply by berndverhofstadt** [reply]
I do think that node-red is a great product! It's very useful in home automation. When it comes to connecting SaaS application, it seems that n8n is more suitable & easier to do the integration. I think they are both great 👍🏻 😄 

E.g. if you want to automate a software flow I would choose n8n (comparable to zapier, IFTTT, Power Automate Flow).
When working with home automation, node-red has some advantages like build in MQTT etc.

Just to be clear, my intention is not to have node-red replaced, I still use it, and it's awesome on DietPi! I discovered DietPi when I was searching for node red on a raspberry pi, I can imagine that people who are looking for n8n will find that it's very easy in combination with DietPi.

**Reply by MichaIng** [reply]
> Just to be clear, my intention is not to have node-red replaced

That is clear, I didn't suspect anything, was just wondering how it compares to Node-RED and what the differences are. As said, indeed the focus of both is a different once. There are overlaps but one does not make the other one redundant (and even if, we have a lot of overlapping software options, choice/concurrence is a great thing, while not always the easiest 😄).

**Comment by Korpip** [comment]
Yeah it seems to have gotten a good community behind it.
